src/main/java/uk/gov/ons/ctp/integration/eqlaunch/crypto/JWSHelper.java

```diff
@@ -105,7 +105,10 @@ public String decode(JWSObject jwsObject, Key key) throws CTPException {
           log.error("Extracted JWS Payload null");
           throw new CTPException(CTPException.Fault.SYSTEM_ERROR, "Extracted JWS Payload null");
         }
-        return payload.toString();
+        // JSON specification allows you to escape forward slash. Payload object String returned
+        // seems
+        // to have this optional escaping. Simply remove so forward slashes aren't escaped.
+        return payload.toString().replace("\\/", "/");
```

Comment:
I understand that this is an attempt to workaround the currently unknown reason why decoding doesn't give back the same data that was originally encoded. I don't think that decode should be attempting a simplistic replacement as it doesn't know if this is really restoring the outgoing data to match the incoming data. 
In an ideal world we'd fix it so that an encode/decode cycle gives you back the same payload that you started with.  If we are living with the quirk then I think it's best to document the quirk and let the caller decide how to handle it.

Reply:
The JSON specification allows the escaping of forward slashes. We use the net.minidev.json.JSONObject to encode a Map into a JSON payload which escapes the forward slashes. When decoded by the python JSON encoder, decoder library it removes all the escaping including for forward slashes.  In extracting the Payload object in Java it leaves the escaping for the forward slashes.  Interestingly, in the unit test using a Jackson Objectmapper to turn the string back into a Map to compare to the input Map instance the escaping on the forward slashes left in the String from the payload is removed. Have removed the replace statement above, though I still personally would prefer to tidy up this output with the replace statement.